examples/platform/silabs/SiWx917/SiWx917DeviceDataProvider.cpp

```diff
@@ -24,12 +26,183 @@
 
 namespace chip {
 namespace DeviceLayer {
-namespace EFR32 {
+namespace SIWx917 {
+#include <string>
+using namespace std;
 
 // using namespace chip::Credentials;
 using namespace chip::DeviceLayer::Internal;
 
-CHIP_ERROR EFR32DeviceDataProvider::GetSetupDiscriminator(uint16_t & setupDiscriminator)
+#ifdef SIWX917_USE_COMISSIONABLE_DATA
+uint8_t SIWx917DeviceDataProvider::WriteBits(uint8_t * bits, uint8_t offset, uint64_t input, uint8_t numberOfBits,
+                                             uint8_t totalPayloadInBits)
+{
+    if ((offset + numberOfBits) > totalPayloadInBits)
+    {
+        return -1;
+    }
+    uint8_t index = offset;
+    offset += numberOfBits;
+    while (input != 0)
+    {
+        if (input & 1)
+        {
+            bits[int(index / 8)] |= (1 << (index % 8));
+        }
+        index++;
+        input >>= 1;
+    }
+    return offset;
+}
+
+// Generation of the payload using the details provided in the DeviceConfig.h
+void SIWx917DeviceDataProvider::generateQrCodeBitSet(uint8_t * payload)
```

Comment:
Again, why are we not using our existing payload generator functionality?  The previous thread claims "address the comments on the latest commit" but the code has not particularly changed, and the questions I asked were not answered.

Reply:
Sorry my bad, When I rebased the branch I lost the commit which was done to address the comments. Pushed the latest commit